File: gin-admin-backend/internal/data/department.go
package data

import (
	"time"
)

type Department struct {
	ID        string    `gorm:"primaryKey;type:varchar(36)" json:"id"`
	ParentID  string    `gorm:"type:varchar(255)" json:"parentId"`
	Name      string    `gorm:"type:varchar(100);not null" json:"name"`
	Sort      int       `gorm:"type:int;default:0" json:"sort"`
	Status    int       `gorm:"type:tinyint;default:1" json:"status"`
	Remark    string    `gorm:"type:varchar(255)" json:"remark"`
	CreatedAt time.Time `json:"createdAt"`
	UpdatedAt time.Time `json:"updatedAt"`
}

func (Department) TableName() string {
	return "sys_department"
}

File: python-admin-backend/app/models/department.py
from sqlalchemy import Column, String, Integer, DateTime, Text
from sqlalchemy.orm import relationship
from sqlalchemy.sql import func
from app.db.database import Base


class Department(Base):
    __tablename__ = "sys_department"

    id = Column(String(36), primary_key=True, index=True)
    parent_id = Column(String(36), nullable=True)
    name = Column(String(100), nullable=False)
    sort = Column(Integer, server_default="0")
    status = Column(Integer, server_default="1")
    remark = Column(Text, nullable=True)
    created_at = Column(DateTime, server_default=func.now())
    updated_at = Column(DateTime, server_default=func.now(), onupdate=func.now())
